### PR TITLE
[PVR] PVR Manager: No need to derive internal job from CJob as all it…

### PR DESCRIPTION
**Pull request by ksooo**
…s members and functionality is not needed here.

A small cleanup that is a side product of #20118 

Runtime-tested on Android and macOS, latest master.

@phunkyfish when you find some time for a review.

### CODE REVIEW

File: xbmc/pvr/PVRManager.cpp [hunk 1]
```diff
@@ -46,10 +46,22 @@ using namespace PVR;
...
+namespace
+{
+
+class CPVRJob
 {
+public:
+  virtual ~CPVRJob() = default;
+
+  virtual bool DoWork() = 0;
+  virtual const std::string GetType() const = 0;
```
Comment (howie-f) [inline xbmc/pvr/PVRManager.cpp:58]:
```suggestion
  virtual const std::string& GetType() const = 0;
```

File: xbmc/pvr/PVRManager.cpp [hunk 2]
```diff
@@ -61,16 +73,18 @@ class CPVRLambdaJob : public CJob
     return true;
   }
 
-  const char* GetType() const override
-  {
-    return m_type.c_str();
-  }
+  const std::string GetType() const override { return m_type; }
```
Comment (howie-f) [inline xbmc/pvr/PVRManager.cpp:76]:
```suggestion
  const std::string& GetType() const override { return m_type; }
```

**Review by howie-f — COMMENTED** [review]
i think const references should be returned.. and i should take a look if a pure virtual base class is appropiate for orphaned dependencies -> DoWork() in https://github.com/xbmc/xbmc/pull/19843

File: xbmc/pvr/PVRManager.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/PVRManager.cpp:58]: No, this way you force any potential importation if that method to store the value to be returned here in a merger variable it even worse a global static variable

File: xbmc/pvr/PVRManager.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/pvr/PVRManager.cpp:76]: Sorry, no.

File: xbmc/pvr/PVRManager.cpp (hunk 1 above)
Comment (howie-f) [reply xbmc/pvr/PVRManager.cpp:58]: makes sense, lesson learned. thx

**Review by phunkyfish — APPROVED** [review]
Makes perfect sense, cleaner and safer.

**Pull request merged**